types/react-syntax-highlighter/index.d.ts

```diff
@@ -9,17 +9,39 @@
 
 type lineTagPropsFunction = (lineNumber: number) => React.HTMLProps<HTMLElement>;
 
+interface Style {
+    [selector: string]: {
+        [property: string]: string;
+    };
+}
+
+interface RendererParams {
+    rows: any;
+    stylesheet?: any;
+    useInlineStyles?: boolean;
+}
+
+type Renderer = (params: RendererParams) => React.ReactNode;
+
 declare module "react-syntax-highlighter" {
     export interface SyntaxHighlighterProps {
         language?: string;
-        style?: any;
-        customStyle?: any;
-        lineProps?: lineTagPropsFunction | React.HTMLProps<HTMLElement>;
+        style?: Style;
+        children?: string;
+        customStyle?: React.CSSProperties;
         codeTagProps?: React.HTMLProps<HTMLElement>;
         useInlineStyles?: boolean;
         showLineNumbers?: boolean;
+        showInlineLineNumbers?: boolean;
         startingLineNumber?: number;
-        lineNumberStyle?: any;
+        lineNumberContainerStyle?: React.CSSProperties;
+        lineNumberStyle?: React.CSSProperties;
+        wrapLines?: boolean;
+        lineProps?: lineTagPropsFunction | React.HTMLProps<HTMLElement>;
+        renderer?: Renderer;
+        PreTag?: any;
+        CodeTag?: any;
+
         [spread: string]: any;
```

Comment:
The `...rest` from the component is mixed in with the props for the `PreTag`.

A more complete definition, without using this `[string]:any` index, could be something like:

```ts
interface SyntaxHighlighterProps<T extends React.ElementType = 'pre'> extends
  React.ComponentPropsWithoutRef<T> {
  // ...
  PreTag?: T
  // ...
}
```